### PR TITLE
[JBPM-5215] Reuse deployment descriptor classes from common jBPM module

### PR DESCRIPTION
**Pull request by elguardian**
Add three classes + the xsd, including testing to the kie-internal-api

### CODE REVIEW

**Comment by elguardian** [comment]
ok to test

**Comment by elguardian** [comment]
rebased

**Comment by bxf12315** [comment]
@elguardian  @cristianonicolai  The issue related to commit https://github.com/kiegroup/jbpm-wb/commit/590e5e52700508491fec084d0e2567794e2dc5b4.
Now, jbpm-wb build is failed.

File: kie-internal/pom.xml [hunk 1]
```diff
@@ -166,12 +167,20 @@
     <dependency>
       <groupId>com.sun.xml.bind</groupId>
       <artifactId>jaxb-xjc</artifactId>
-      <scope>provided</scope>
+       <scope>compile</scope>
+       <optional>true</optional>
```
Comment (mswiderski) [inline kie-internal/pom.xml]: @MarianMacik how about these dependencies in context of JDK 11 work you're doing?

Comment (MarianMacik) [reply kie-internal/pom.xml]: Hi, I am doing things mostly regarding APIs, so implementations themselves should be good.

**Comment by cristianonicolai** [comment]
Jenkins execute full downstream build

File: kie-internal/pom.xml (hunk 1 above)
Comment (mswiderski) [reply kie-internal/pom.xml]: @elguardian why do we need xjc?

Comment (elguardian) [reply kie-internal/pom.xml]: @mswiderski testing. they fail during unmarshalling if that dependency is not there.

Comment (mswiderski) [reply kie-internal/pom.xml]: but does it have to be compile scope?

Comment (elguardian) [reply kie-internal/pom.xml]: fixed to provided

**Pull request merged**